rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/savedFilters.spec.ts

```diff
@@ -20,143 +21,164 @@ jest.mock("../../../../library/stores/ActivityFilterStore", () => ({
-        console.log("Before delete:", wrapper.vm.filters);
-        // Mock the deleteFilter method
-        wrapper.vm.deleteFilter = (filterName) => {
-          const index = wrapper.vm.filters.findIndex(
-            (filter) => filter.filterName === filterName,
-          );
-          if (index !== -1) {
-            wrapper.vm.filters.splice(index, 1);
-          }
-        };
-        await wrapper.vm.deleteFilter(wrapper.vm.filters[0].filterName);
+      const initialCount = wrapper.findAll(
+        '[data-test-id="filter-item"]'
+      ).length;
+      expect(initialCount).toBeGreaterThan(0);
+      const deleteButton = wrapper.find('[data-test-id="delete-filter-btn"]');
+      if (deleteButton.exists()) {
+        await deleteButton.trigger("click");
         await wrapper.vm.$nextTick();
-        console.log("After delete:", wrapper.vm.filters);
-
-        expect(wrapper.vm.filters.length).toBeLessThan(initialCount);
-      } else {
-        throw new Error("No filters available to delete");
+        const finalCount = wrapper.findAll(
+          '[data-test-id="filter-item"]'
+        ).length;
+        expect(removeFilterSpy).not.toHaveBeenCalled();
+        expect(finalCount).toBe(initialCount);
       }
     });
-
-    it("shows 'No filters available' when filters array is empty", async () => {
-      wrapper.vm.filters = [];
-      await wrapper.vm.$nextTick();
-
-      const noFiltersMessage = wrapper
-        .find('[data-test-id="no-filters-message"]')
-        .text();
-      expect(noFiltersMessage).toContain("No filters available");
+    it("shows an empty dropdown when filters array is empty", async () => {
+      const wrapper = mountSavedFilters({ filters: [] });
+      const dropdownItems = wrapper.findAll('[data-test-id="filter-item"]');
+      expect(dropdownItems.length).toBe(0);
     });
     it("updates visible elements when 'query.filterName' changes", async () => {
-      await wrapper.setProps({ query: { filterName: "Updated Filter" } });
+      const wrapper = mountSavedFilters({
+        query: { filterName: "Updated Filter" },
+      });
       expect(wrapper.find('[data-test-id="filter-name"]').text()).toBe(
-        "Updated Filter",
+        "Updated Filter"
       );
     });
+
+    it("updates filters after successful deletion", async () => {
+      const wrapper = mountSavedFilters();
+      const initialCount = wrapper.findAll(
+        '[data-test-id="filter-item"]',
+      ).length;
+      // Mock the $confirm to return a resolved promise
+      wrapper.vm.$confirm = () => Promise.resolve();
+      // Mock ActivityFilterStore.removeFilter
+      ActivityFilterStore.prototype.removeFilter = jest
+        .fn()
+        .mockImplementation((filterName) => {
+          const index = wrapper.vm.filters.findIndex(
+            (filter) => filter.filterName === filterName
+          );
+          if (index !== -1) {
+            wrapper.vm.filters.splice(index, 1);
+          }
+        });
```

Comment:
Might be needed to tweak this mock as well, to either match the format of the editProjectFileService in `EditProjectFile.spec.ts` OR follow one of the approaches described here: https://www.mikeborozdin.com/post/changing-jest-mocks-between-tests